girder/api/rest.py

```diff
@@ -1011,6 +1011,20 @@ def _matchRoute(self, method, path):
                     break
             else:
                 return route, handler, wildcards
+        # Handle routes with variable arguments
```

Comment:
This new `+` syntax appears to be a new invention for Girder (as opposed to something inherited from CherryPy). Should we add developer documentation? Should we add more explicit testing?

Reply:
It also looks like there's some logic here (e.g. support for both `:` and `+` in a route) for cases which are not covered implicitly by the use of `resource/path/download/+path`. Do we have any validation that it works? The addition of this new code significantly increases the cyclomatic complexity of `_matchRoute`.